keps/sig-storage/3314-csi-volume-snapshot-delta/README.md

```diff
@@ -0,0 +1,1441 @@
+details). For now, we leave it here.
+-->
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+###### What are other known failure modes?
+
+<!--
+For each of them, fill in the following information by copying the below template:
+  - [Failure mode brief description]
+    - Detection: How can it be detected via metrics? Stated another way:
+      how can an operator troubleshoot without logging into a master or worker node?
+    - Mitigations: What can be done to stop the bleeding, especially for already
+      running user workloads?
+    - Diagnostics: What are the useful log messages and their required logging
+      levels that could help debug the issue?
+      Not required until feature graduated to beta.
+    - Testing: Are there any tests for failure mode? If not, describe why.
+-->
+
+###### What steps should be taken if SLOs are not being met to determine the problem?
+
+## Implementation History
+
+<!--
+Major milestones in the lifecycle of a KEP should be tracked in this section.
+Major milestones might include:
+- the `Summary` and `Motivation` sections being merged, signaling SIG acceptance
+- the `Proposal` section being merged, signaling agreement on a proposed design
+- the date implementation started
+- the first Kubernetes release where an initial version of the KEP was available
+- the version of Kubernetes where the KEP graduated to general availability
+- when the KEP was retired or superseded
+-->
+
+## Drawbacks
+
+<!--
+Why should this KEP _not_ be implemented?
+-->
+
+## Alternatives
+
+<!--
+What other approaches did you consider, and why did you rule them out? These do
+not need to be as detailed as the proposal, but should include enough
+information to express the idea and why it was not acceptable.
+-->
+
+The previous alternate design which involves generating and returning a callback
+endpoint to the caller has been superceded by the aggregation extension
+mechanism described in this KEP. The aggregation extension design provides a
+tighter integration with the Kubernetes API server, enabling the re-use of the
+existing Kubernetes machinery of GVR and GVK binding, URL registration and
+delegated authentication and authorization.
+
+Another alternative that allows user to provide the callback URL to send the
+CBT response payload to has also been rejected due to concerns around the lack
+of control over the authenticity of the remote URLs.
+
+Some consideration was given to associating the initialization of CBT requests
+to the `LIST` action, instead of the `CREATE` action, to match the "retrieve a
+list of CBT entries" semantics. This KEP decided to implement the `CREATE`
+action as its allows us to encapsulate the CBT request parameters as the
+structured `VolumeSnapshotDelta` resources. The proposed API also ensures that
+the `CREATE` action provides user with the pagination control over the amount of
+data returned.
+
```

Comment:
Please add an alternative with persisting `VolumeSnapshotDelta` objects in etcd + adding a subresource that would list the actual diffs. The aggregated API server would be just a HTTP(S) proxy between a backup SW / HTTP client and CSI CBT sidecar. CSI CBT sidecar SW could stream CBT records without any translation.

Reply:
I updated the 'Alternatives' section to include this.